doc/spec/skupper-cli-spec.md

```diff
@@ -0,0 +1,287 @@
+# Skupper CLI spec
+
+## Hello World, briefly
+
+~~~ console
+# Get the CLI
+
+$ curl https://skupper.io/install.sh | curl
+
+# West
+
+$ export KUBECONFIG=~/.kube/config-west
+$ kubectl apply -f https://skupper.io/install.yaml
+$ kubectl create deployment frontend --image quay.io/skupper/hello-world-frontend
+
+$ skupper site create --ingress loadbalancer
+$ skupper token create ~/token.yaml
+$ skupper listener create backend --host backend --port 8080
+
+# East
+
+$ export KUBECONFIG=~/.kube/config-east
+$ kubectl apply -f https://skupper.io/install.yaml
+$ kubectl create deployment backend --image quay.io/skupper/hello-world-backend --replicas 3
+
+$ skupper site create
+$ skupper link create ~/token.yaml
+$ skupper connector create backend --workload deployment/backend --port 8080
+~~~
+
+## Philosophy
+
+The Skupper CLI in version 2 is deliberately a light layer on top of
+the standard Skupper customer resources.  Its job in short is to
+render the YAML resource, submit it to the platform, and wait for the
+desired result.
+
+The `create` and `set` operations in particular are meant to provide a
+convenient and CLI-conventional interface, as an alternative to
+writing YAML by hand.
+
+In general, the operations block until the user's desired outcome is
+achieved.  If the user wants something asynchronous, they can use
+`kubectl` commands.
+
+## Example site operations
+
+~~~ console
+$ skupper site create --name west --ingress loadbalancer
+Waiting for status...
+Waiting for ingress...
+Site "west" is ready
+
+$ skupper site get
+NAME   STATUS   INGRESS
+west   Ready    loadbalancer
+
+$ skupper site get -o yaml
+apiVersion: v2alpha1
+kind: Site
+[...]
+
+$ skupper site set --ingress route
+Waiting for ingress...
+Site "west" is ready
+
+$ skupper site delete
+Waiting for site "west" to delete...
+Site "west" is deleted
+~~~
+
+## Example token operations
+
+~~~ console
+$ skupper token create token.yaml
+Token file created at token.yaml
+The token expires after 1 use or after 15 minutes
+~~~
+
+## Example link operations
+
+~~~ console
+$ skupper link create token.yaml
+Waiting for link "west" to become active...
+Link "west" is active
+You can now delete token.yaml
+
+$ skupper link get
+NAME    STATUS   COST
+south   Error    10
+west    Active   1
+
+$ skupper link get west
+NAME   STATUS   COST
+west   Active   1
+
+$ skupper link get west -o yaml
+apiVersion: v2alpha1
+kind: Link
+[...]
+
+$ skupper link delete west
+Waiting for link "west" to delete...
+Link "west" is deleted
+~~~
+
+## Example listener operations
+
+~~~ console
+$ skupper listener create database --host database --port 5432
+Waiting for listener...
+Listener "database" is ready
+
+$ skupper listener get
+NAME       ROUTING-KEY   HOST       PORT
+payments   payments      payments   8080
+database   database      database   5432
+
+$ skupper listener get database -o yaml
+apiVersion: v2alpha1
+kind: Listener
+[...]
+
+$ skupper listener set database --port 5431
```

Comment:
To me, this looks a little odd. Have we considered 'update' instead of 'set'?

Reply:
@ssorj these comments are really for you I think? (I have no objection to committing the doc as is. This was my first sight of it and so I was commenting here for convenience.